test/0.8.9/sepolia-deposit-adapter.test.js

```diff
@@ -0,0 +1,142 @@
+const { contract, ethers } = require('hardhat')
+const { assert } = require('../helpers/assert')
+
+const { EvmSnapshot } = require('../helpers/blockchain')
+
+// To run Sepolia Deposit Adapter tests:
+// HARDHAT_FORKING_URL=<rpc url> HARDHAT_CHAIN_ID=11155111 npx hardhat test --grep "SepoliaDepositAdapter"
+contract('SepoliaDepositAdapter', ([deployer]) => {
+  let depositAdapter
+  let snapshot
+  let bepoliaToken
+  // const sepoliaDepositAdapterContract = '0x899e45316FaA439200b36c7d7733192530e3DfC0'
+  const sepoliaDepositContract = '0x7f02C3E3c98b133055B8B348B2Ac625669Ed295D'
+  const EOAddress = '0x6885E36BFcb68CB383DfE90023a462C03BCB2AE5'
+  const bepoliaTokenHolder = EOAddress
+  // const log = console.log
+  const log = () => {}
+
+  before('deploy lido with dao', async function () {
+    const { chainId } = await ethers.provider.getNetwork()
+    if (chainId !== 11155111) {
+      return this.skip()
+    }
+
+    // depositAdapter = await SepoliaDepositAdapter.at(sepoliaDepositAdapterContract)
+    depositAdapter = await ethers.deployContract('SepoliaDepositAdapter', [sepoliaDepositContract])
+    log('depositAdapter address', depositAdapter.address)
+
+    bepoliaToken = await ethers.getContractAt('ISepoliaDepositContract', sepoliaDepositContract)
+
+    const result = await depositAdapter.name()
+    assert.equals(result, 'Sepolia deposit contract token')
+
+    snapshot = new EvmSnapshot(ethers.provider)
+    await snapshot.make()
+  })
+
+  afterEach(async () => {
+    await snapshot.rollback()
+  })
+
+  describe('SepoliaDepositAdapter Logic', () => {
```

Comment:
I guess we might need to test:
- `receive` method
- events to be emitted, esp `DepositEvent`

Reply:
It's the only item left. I can't detect `DepositEvent` was sent from the original contract. Still figuring this out.

Reply:
ok. this solved as well